File: src/mode/host-runtime.mjs
import { GatedLoopError } from '../core/errors.mjs';

export const AGENT_RUNTIME_PATTERN = /^[a-z][a-z0-9._-]{0,63}$/;

export function isAgentRuntime(value) {
  return typeof value === 'string' && AGENT_RUNTIME_PATTERN.test(value);
}

export function normalizeHostRuntime(hostRuntime) {
  if (hostRuntime === undefined || hostRuntime === null) return undefined;
  if (!isAgentRuntime(hostRuntime)) {
    throw new GatedLoopError('HOST_RUNTIME_INVALID', 'hostRuntime must be a safe lowercase Agent identifier');
  }
  return hostRuntime;
}

export function requireHostRuntime(hostRuntime) {
  const normalized = normalizeHostRuntime(hostRuntime);
  if (!normalized) {
    throw new GatedLoopError('HOST_RUNTIME_REQUIRED', 'A writing workflow requires a --host-runtime Agent identifier');
  }
  return normalized;
}

File: skills/layered-delivery/scripts/hdg/host_runtime.py
from __future__ import annotations

import re

from .errors import GatedLoopError


AGENT_RUNTIME_PATTERN = re.compile(r"^[a-z][a-z0-9._-]{0,63}$")


def is_agent_runtime(value: object) -> bool:
    return isinstance(value, str) and bool(AGENT_RUNTIME_PATTERN.fullmatch(value))


def is_claude_runtime(value: object) -> bool:
    return bool(
        is_agent_runtime(value)
        and (value == "claude" or value.startswith("claude-") or value.startswith("claude."))
    )


def require_host_runtime(value: object) -> str:
    if not is_agent_runtime(value):
        raise GatedLoopError(
            "HOST_RUNTIME_REQUIRED" if value is None else "HOST_RUNTIME_INVALID",
            "A writing workflow requires a --host-runtime Agent identifier"
            if value is None
            else "hostRuntime must be a safe lowercase Agent identifier",
        )
    return value
